spacy_experimental/coref/coref_component.py

```diff
@@ -0,0 +1,395 @@
+from typing import Iterable, Tuple, Optional, Dict, Callable, Any, List
+import warnings
+
+from thinc.types import Floats2d, Floats3d, Ints2d
+from thinc.api import Model, Config, Optimizer
+from thinc.api import set_dropout_rate, to_categorical
+from itertools import islice
+from statistics import mean
+import srsly
+
+from spacy.pipeline.trainable_pipe import TrainablePipe
+from spacy.language import Language
+from spacy.training import Example, validate_examples, validate_get_examples
+from spacy.errors import Errors
+from spacy.tokens import Doc
+from spacy.vocab import Vocab
+from spacy.util import from_disk, from_bytes
+
+from .coref_util import create_gold_scores, MentionClusters, create_head_span_idxs
+from .coref_util import get_clusters_from_doc, get_predicted_clusters
+from .coref_util import DEFAULT_CLUSTER_PREFIX
+
+from .coref_scorer import score_coref_clusters
+
+
+default_config = """
+[model]
+@architectures = "spacy-experimental.Coref.v1"
+distance_embedding_size = 20
+hidden_size = 1024
+depth = 1
+dropout = 0.3
+antecedent_limit = 50
+antecedent_batch_size = 512
+
+[model.tok2vec]
+@architectures = "spacy.Tok2Vec.v2"
+
+[model.tok2vec.embed]
+@architectures = "spacy.MultiHashEmbed.v1"
+width = 64
+rows = [2000, 2000, 1000, 1000, 1000, 1000]
+attrs = ["ORTH", "LOWER", "PREFIX", "SUFFIX", "SHAPE", "ID"]
```

Comment:
Don't use `ID`, it's just the row number in the vector table for default vectors, and `NORM` is usually better than `ORTH`. I suspect that you don't need `NORM` and `LOWER`, but you'd have to check experimentally. It probably also makes sense to increase the rows for `NORM` and decrease the rows for `PREFIX`.

This would be a bit closer to the normal trained pipeline tok2vec defaults, but keeps `LOWER`:

```suggestion
attrs = ["NORM", "LOWER", "PREFIX", "SUFFIX", "SHAPE"]
rows = [5000, 2500, 1000, 2500, 2500]
```

Reply:
Thanks! I don't think those values were carefully chosen. 

In practice we've been developing this on the assumption it'll primarily be used with Transformers, so these tok2vec settings haven't been checked much. We can give them a proper test.